### PR TITLE
Add metrics push gateway for supervisor

### LINKED ISSUE #9515: [ide-proxy] expose endpoints to collect workspace metrics
Since workspaces are ephemeral by nature we cannot poll them to collect metrics. Instead workspaces should increase counters on a new long living component which can expose them to prometheus. We could use IDE proxy for that.

### PR DESCRIPTION
**Pull request by iQQBot**
## Description

This PR allow prometheus to scrape metrics from supervisor, and allow IDE push metrics to supervisor

need manual add this CRD to clusters, this is another part for https://github.com/gitpod-io/observability
```yaml
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: workspace
spec:
  selector:
    matchLabels:
      component: workspace
  podMetricsEndpoints:
  - port: supervisor
```

## Related Issue(s)

Fixes #9515

## How to test

1. start a workspace with `latest` vscode
2. try type something in order to active code suggestion
3. look at prometheus

image
image
image

## Release Notes

```release-note
Adding metrics reporting to the IDE
```

## Documentation

- [x] /werft no-preview

### CODE REVIEW

File: WORKSPACE.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ defaultArgs:
   jbMarketplacePublishTrigger: "false"
   publishToJBMarketplace: true
   localAppVersion: unknown
-  codeCommit: 30d694d1f41945af7f8d757d3ddf176b283752cb
+  codeCommit: c8ec2992780f3346b45cd441df19ce79929023a6
```
Comment (akosyakov) [inline WORKSPACE.yaml:10]: @jeanp413 is it fine to merge like that?

Comment (akosyakov) [reply WORKSPACE.yaml:10]: does not matter, Pudong is going to remove it

**Review by sagor999 — APPROVED** [review]
workspace portion LGTM

**Comment by akosyakov** [comment]
@iQQBot Should there be changes in ws-proxy which prohibit access to metrics endpoint by anyone else but prometheus?

**Pull request merged**